onnxt5/api.py

```diff
@@ -42,11 +49,29 @@ def run_embeddings_text(encoder, decoder, tokenizer, prompt):
 
     return encoder_output, decoder_output
 
-def download_generation_model(output_dir, object):
-    url = f'https://t5-onnx-models.s3.amazonaws.com/{object}'
-    r = requests.get(url, allow_redirects=True)
-    downloaded_file_path = os.path.join(output_dir, object)
-    open(downloaded_file_path, 'wb').write(r.content)
-    tar = tarfile.open(downloaded_file_path, "r:gz")
-    tar.extractall(path=output_dir)
-    tar.close()
+
+def _progress(t):
+    def inner(bytes_amount):
+        t.update(bytes_amount)
+
+    return inner
+
+
+def _download_generation_model(path):
+    bucket = 't5-onnx-models'
+    key = str(path.name)
+
+    s3 = boto3.client('s3')
+
+    # Gets size of object in bytes
+    response = s3.head_object(Bucket=bucket, Key=key)
+    size = response['ContentLength']
+
+    # Downloads from S3, reporting progress in bytes to a tqdm progress bar. Units are in bytes. Setting disable to None
+    # causes tqdm to check whether the process is attached to a terminal and disable progress bar output if not.
+    with tqdm(total=size, unit='B', unit_scale=True, desc=key, disable=None) as t:
+        s3.download_file(Bucket=bucket, Key=key, Filename=str(path), Callback=_progress(t))
```

Comment:
If you want me to change this portion to use `request` instead of `boto3`, I can do that.